packages/monitoring-system/data-processor/src/data-processors/cloud-functions-data-processor.ts

```diff
@@ -12,14 +12,67 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 //
-import {DataProcessor} from './data-processor-abstract';
+import {DataProcessor, ProcessorOptions} from './data-processor-abstract';
+import {GCF, CloudFunction} from 'googleapis-nodejs-functions';
```

Comment:
So originally I wanted to use the [`@google-cloud/functions`](https://github.com/googleapis/nodejs-functions) library but that was [having some issues](https://github.com/googleapis/nodejs-functions/issues/13). 

@sofisl suggested using this library and it seemed to do the trick.